Tests/XCUITests/HistoryTests.swift

```diff
@@ -103,215 +98,214 @@ class HistoryTests: BaseTestCase {
         navigator.goto(HistoryRecentlyClosed)
+        waitForExistence(app.tables["Recently Closed Tabs List"])
+        XCTAssertTrue(app.tables.cells.staticTexts[bookOfMozilla["label"]!].exists)
+    }
 
-        // The Closed Tabs list should contain the info of the website just closed
-        waitForExistence(app.tables["Recently Closed Tabs List"], timeout: 3)
-        XCTAssertTrue(app.tables.cells.staticTexts[closedWebPageLabel].exists)
+    func testRecentlyClosedPrivateMode() {
+        openBookOfMozilla()
 
-        navigator.goto(HomePanelsScreen)
+        closeFirstTabByX()
 
-        // This option should be enabled on private mode too
+        // Toggle to private mode. The closed tab is listed in "Recently Closed Tabs List"
+        navigator.nowAt(HomePanelsScreen)
         navigator.toggleOn(userState.isPrivate, withAction: Action.TogglePrivateMode)
-        navigator.performAction(Action.OpenNewTabFromTabTray)
-        navigator.nowAt(NewTabScreen)
-        navigator.performAction(Action.CloseURLBarOpen)
-        navigator.nowAt(NewTabScreen)
-        navigator.goto(BrowserTabMenu)
         navigator.goto(HistoryRecentlyClosed)
         waitForExistence(app.tables["Recently Closed Tabs List"])
+        XCTAssertTrue(app.tables.cells.staticTexts[bookOfMozilla["label"]!].exists)
     }
 
-    func testClearRecentlyClosedHistory() {
-        // Open the default website
-        navigator.openURL(path(forTestPage: "test-mozilla-book.html"))
-        navigator.nowAt(BrowserTab)
+    func testRemoveAllTabsButton() {
+        openBookOfMozilla()
+
+        // Tap "Remove All Tabs" instead of close the tab individually
         waitForTabsButton()
         navigator.goto(TabTray)
         navigator.performAction(Action.AcceptRemovingAllTabs)
-        navigator.performAction(Action.CloseURLBarOpen)
-        navigator.nowAt(NewTabScreen)
-        navigator.goto(BrowserTabMenu)
-        navigator.goto(HistoryRecentlyClosed)
-        // Once the website is visited and closed it will appear in Recently Closed Tabs list
-        waitForExistence(app.tables["Recently Closed Tabs List"])
-        XCTAssertTrue(app.tables.cells.staticTexts[closedWebPageLabel].exists)
 
-        navigator.goto(HomePanelsScreen)
-        navigator.nowAt(NewTabScreen)
+        // The closed tab is *not* listed in "Recently Closed Tabs List"
+        navigator.nowAt(HomePanelsScreen)
+        navigator.goto(LibraryPanel_History)
+        waitForExistence(app.tables[HistoryPanelA11y.tableView])
+        XCTAssertTrue(app.tables[HistoryPanelA11y.tableView].staticTexts[emptyRecentlyClosedMesg].exists)
+        XCTAssertFalse(app.tables.cells.staticTexts[bookOfMozilla["label"]!].exists)
+    }
+
+    func testClearRecentlyClosedHistory() {
+        openBookOfMozilla()
 
-        // Go to settings and clear private data
+        closeFirstTabByX()
+
+        // Clear all private data via the settings
+        navigator.nowAt(HomePanelsScreen)
```

Comment:
As commented on other tests, you have to add these lines after this:
```
waitForExistence(app.buttons["urlBar-cancel"])
navigator.performAction(Action.CloseURLBarOpen)
navigator.nowAt(NewTabScreen)
```
So that the `goto(ClearPrivateData)` works